Initial CONTRIBUTING draft based on https://bttger.github.io/contributing-gen-web.

demize opened the pull request:
Mainly modified the security reporting section to point users to the new Github built-in feature.

There's still a fair bit to be done here, a few sections marked TODO (and obviously blank in the render):

- [x] Clear first contribution guidelines
- [ ] Clear documentation guidelines
- [ ] Clear style guides:
  - [x] JS style (what formatter/linter do we want to use? we should adopt one)
  - [ ] Branch naming style
  - [x] Commit messages (specifically called out in the template)
- [x] "Join The Project Team"... probably just remove this section? We can add it back in if we need to.

Not directly related, but two additional things we may want to add while we're working on this:

- [x] Pull request template
- [x] Issue template (or two; one for issues, one for feature requests)

liquidnya commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,139 @@
+
+> And if you like the project, but just don't have time to contribute, that's fine. There are other easy ways to support the project and show your appreciation, which we would also be very happy about:
+> - Star the project
+> - Tweet about it
```
With what is going on with Twitter at the moment;
I wonder if rephrasing this to `Talk about it on your favorite social media site` or something like that would be a good idea.

demize commented:
Well this has become a bit of an omnibus PR...

I've set up eslint, formatted the code and fixed all the errors it pointed out (we had a couple variables and consts that were just never used).

I added a bug report and enhancement request issue template, and a PR template.

And I think I've effectively finished the CONTRIBUTING document as well... I'm not confident I like all the parts of it, but I think at this point it's best just to get someone else to take a look 😅

liquidnya commented on .editorconfig:
```diff
@@ -0,0 +1,14 @@
+trim_trailing_whitespace = true
+charset = utf-8
+
+# JS, YAML files
+[*.{js, yml}]
+intent_style = space
+indent_size = 2
```
should we also cover `.ts` and `.json` files too?

liquidnya commented on .editorconfig:
We could also consider adding a `.gitattributes` file.
See <https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings>.

liquidnya reviewed (approved):
Looks good to me!
Just 2 comments which are not that important and could be done in a separate PR too (or not be done at all).

demize commented:
> Looks good to me! Just 2 comments which are not that important and could be done in a separate PR too (or not be done at all).

Both are good points that I think we should worry about later. For now I want to get this merged so we can worry about rebasing your branch, since that's gonna be a huge pain...
